src/middleware/statement.validation.middleware.ts

```diff
@@ -0,0 +1,43 @@
+import { NextFunction, Request, Response } from "express";
+import { isActiveFeature } from "../utils/feature.flag";
+import { checkActiveBOExists, getApplicationData } from "../utils/application.data";
+import {
+  FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION,
+  UPDATE_CHECK_YOUR_ANSWERS_URL,
+  UPDATE_REVIEW_STATEMENT_URL
+} from '../config';
+import { BeneficialOwnerStatementKey, BeneficialOwnersStatementType } from "../model/beneficial.owner.statement.model";
+import { ApplicationData } from "../model";
+import { ErrorMessages } from "../validation/error.messages";
+import { Session } from "@companieshouse/node-session-handler";
+
+export const hasValidStatements = (req: Request, res: Response, next: NextFunction) => {
+  const errorList: string[] = [];
+  const appData: ApplicationData = getApplicationData(req.session as Session);
+
+  if (
+    isActiveFeature(FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION) &&
+      !checkStatementsValid(appData, errorList)
+  ) {
+    req['statementErrorList'] = errorList;
+    return next();
+  }
+
+  const redirectUrl = appData.update?.no_change ? UPDATE_REVIEW_STATEMENT_URL : UPDATE_CHECK_YOUR_ANSWERS_URL;
+  return res.redirect(redirectUrl);
+};
+
+const validateIdentifiedBOsStatement = (appData: ApplicationData, errorList: string[]): boolean => {
```

Comment:
Do we want to move this to somewhere outside of the middleware? For reuse on the nav ticket?

Reply:
Discussed, will do in later ticket.